esp-wifi/src/wifi/utils.rs

```diff
@@ -44,7 +44,7 @@ fn setup_iface<'a, MODE: WifiDeviceMode>(
 /// storage.
 pub fn create_network_interface<'a, 'd, MODE: WifiDeviceMode>(
     inited: &EspWifiInitialization,
-    device: impl crate::hal::peripheral::Peripheral<P = crate::hal::peripherals::WIFI> + 'd,
+    device: &'d mut (impl crate::hal::peripheral::Peripheral<P = crate::hal::peripherals::WIFI> + 'd),
```

Comment:
```suggestion
    device: impl crate::hal::peripheral::Peripheral<P = crate::hal::peripherals::WIFI> + 'd,
```

This should be equivalent